dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -323,18 +343,34 @@ await foreach (Completions completions in response.ConfigureAwait(false))
         for (int requestIndex = 1; ; requestIndex++)
         {
             // Make the request.
-            var responseData = (await RunRequestAsync(() => this.Client.GetChatCompletionsAsync(chatOptions, cancellationToken)).ConfigureAwait(false)).Value;
-            this.CaptureUsageDetails(responseData.Usage);
-            if (responseData.Choices.Count == 0)
+            ChatCompletions responseData;
+            IEnumerable<OpenAIChatMessageContent> responseContent;
+            using (var activity = ModelDiagnostics.StartCompletionActivity(this.Endpoint, this.DeploymentOrModelName, ModelProvider, chat, executionSettings))
             {
-                throw new KernelException("Chat completions not found");
+                try
+                {
+                    responseData = (await RunRequestAsync(() => this.Client.GetChatCompletionsAsync(chatOptions, cancellationToken)).ConfigureAwait(false)).Value;
+                    this.CaptureUsageDetails(responseData.Usage);
+                    if (responseData.Choices.Count == 0)
+                    {
+                        throw new KernelException("Chat completions not found");
```

Comment:
same here, let's record the properties we can record even if there are no choices